core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java

```diff
@@ -86,111 +75,164 @@ public void setHome() {
+        Infrastructure infrastructure = productTestPlan.getInfrastructureMap().get(DEPLOYMENT_PATTERN);
+        Assert.assertNotNull(infrastructure);
+        Assert.assertTrue(DEPLOYMENT_PATTERN.equals(infrastructure.getName()));
+        Assert.assertTrue("K8S".equals(infrastructure.getClusterType().name()));
+        Assert.assertTrue("OPENSTACK".equals(infrastructure.getProviderType().name()));
+        Assert.assertTrue("DOCKER_CONTAINERS".equals(infrastructure.getInstanceType().name()));
+        Assert.assertNotNull(infrastructure.getDatabase());
+        Assert.assertTrue(Database.DatabaseEngine.MYSQL.equals(infrastructure.getDatabase().getEngine()));
+        Assert.assertTrue("5.7".equals(infrastructure.getDatabase().getVersion()));
+        Assert.assertNotNull(infrastructure.getOperatingSystem());
+        Assert.assertTrue("Ubuntu".equals(infrastructure.getOperatingSystem().getName()));
+        Assert.assertTrue("17.04".equals(infrastructure.getOperatingSystem().getVersion()));
+        Assert.assertNotNull(infrastructure.getSecurityProperties());
+        Assert.assertNotNull(infrastructure.getScripts());
+        Assert.assertTrue(infrastructure.getScripts().size() == 2);
+    }
 
+    @Test (dependsOnMethods = "parseProductTestPlanTest")
+    public void executeProductTestPlanTestWithUnsupportedProvider() throws UnsupportedProviderException,
+            TestGridException, InfrastructureProviderInitializationException {
         PowerMockito.mockStatic(InfrastructureProviderFactory.class);
-
-        Mockito.when(InfrastructureProviderFactory.getInfrastructureProvider(infrastructure)).thenReturn(provider);
-        Mockito.when(provider.createInfrastructure(infrastructure, scenarioLocation)).thenReturn(deployment);
-        Mockito.when(provider.removeInfrastructure(infrastructure, scenarioLocation)).thenReturn(true);
-        Mockito.when(provider.canHandle(infrastructure)).thenReturn(true);
-        Mockito.when(provider.removeInfrastructure(infrastructure, scenarioLocation)).thenReturn(true);
-
-        Mockito.when(deployer.deploy(testPlan.getDeployment())).thenReturn(deployment);
-        Mockito.when(testEngine.runScenario(testScenario, scenarioLocation, deployment)).thenReturn(true);
-
-        try {
-            PowerMockito.whenNew(TestEngineImpl.class).withNoArguments().thenReturn(testEngine);
-            PowerMockito.whenNew(TestReportEngineImpl.class).withNoArguments().thenReturn(testReportEngine);
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-
-        new TestGridMgtServiceImpl().executeProductTestPlan(productTestPlan);
-        Assert.assertTrue(true);
+        Mockito.when(InfrastructureProviderFactory.getInfrastructureProvider(Mockito.anyObject()))
+                .thenThrow(new UnsupportedProviderException());
+        Assert.assertTrue(new TestGridMgtServiceImpl().executeProductTestPlan(productTestPlan));
     }
+
+//    @Test
```

Comment:
We should use testng.xml to comment out tests (we can handle this via another issue)

Reply:
Agreed. Will be fixed with #54